microsoft/testsuites/mshv/mshv_root_tests.py

```diff
@@ -0,0 +1,198 @@
+import time
+from pathlib import Path
+from typing import Any, Dict
+
+from assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import Dmesg, Free, Ls, Lscpu, QemuImg, Wget
+from lisa.util import SkippedException
+from microsoft.testsuites.mshv.cloud_hypervisor_tool import CloudHypervisor
+
+
+@TestSuiteMetadata(
+    area="mshv",
+    category="",
+    description="""
+    This test suite contains tests that are meant to be run on the
+    Microsoft Hypervisor (MSHV) root partition.
+    """,
+)
+class MshvHostTestSuite(TestSuite):
+    CONFIG_VARIABLE = "mshv_vm_create_stress_configs"
+    VM_CREATE_STRESS_DEFAULT_CONFIGS = [
+        {
+            "times": 25,
+            "cpus_per_vm": 1,
+            "mem_per_vm_mb": 1024,
+        }
+    ]
+
+    HYPERVISOR_FW_NAME = "hypervisor-fw"
+    DISK_IMG_NAME = "vm_disk_img.raw"
+
+    def before_suite(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        if not node.tools[Ls].path_exists("/dev/mshv", sudo=True):
+            raise SkippedException("This suite is for MSHV root partition only")
+        working_path = node.get_working_path()
+        node.tools[Wget].get(
+            "https://github.com/cloud-hypervisor/rust-hypervisor-firmware/releases/download/0.4.1/hypervisor-fw",  # noqa: E501
+            file_path=str(working_path),
+            filename=self.HYPERVISOR_FW_NAME,
+        )
+        node.tools[Wget].get(
+            "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img",  # noqa: E501
+            file_path=str(working_path),
+            filename=f"{self.DISK_IMG_NAME}.img",
+        )
+        node.tools[QemuImg].convert(
+            "qcow2",
+            str(working_path / f"{self.DISK_IMG_NAME}.img"),
+            "raw",
+            str(working_path / self.DISK_IMG_NAME),
+        )
+
+    @TestCaseMetadata(
+        description="""
+            Stress the MSHV virt stack by repeatedly creating and destroying
+            multiple VMs in parallel. By default creates VMs with 1 vCPU and
+            1 GiB of RAM each. Number of VMs createdis equal to the number of
+            CPUs available on the host. By default, the test is repeated 25
+            times. All of these can be configured via the variable
+            "mshv_vm_create_stress_configs" in the runbook.
+        """,
+        priority=3,
+    )
+    def verify_mshv_stress_vm_create(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any],
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+    ) -> None:
+        if self.CONFIG_VARIABLE in variables:
+            configs = variables[self.CONFIG_VARIABLE]
```

Comment:
Give each variable a default value, so, the test case can run without any configuration. And it's better to expand each variable, so it's easy to set some of them, instead of all each time.

Reply:
The test case can be run without any configuration as-is. I have now implemented a default value for each parameter so that some of them can be skipped.